protocol-units/dispute/src/MCR.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Output, OutputLib, Receipt, ReceiptClaim, ReceiptClaimLib, IRiscZeroVerifier, SystemExitCode, ExitCode} from "./IRiscZeroVerifier.sol";
+import {SafeCast} from "openzeppelin-contracts/contracts/utils/math/SafeCast.sol";
+import "forge-std/console2.sol";
+
+contract MCR {
+    using ReceiptClaimLib for ReceiptClaim;
+    using OutputLib for Output;
+    using SafeCast for uint256;
+
+    struct Validator {
+        bool isRegistered;
+        uint256 stake;
+    }
+
+    struct OptimisticCommitment {
+        bytes32 blockHash;
+        uint256 stateCommitmentCount;
+        uint256 totalStake;
+        bool isAccepted;
+    }
+
+    struct Dispute {
+        bytes32 blockHash;
+        uint256 timestamp;
+        DisputeState state;
+    }
+
+    struct Proof {
+        bytes32 blockHash;
+        bool isValid;
+        bool exists;
+    }
+
+    enum DisputeState {
+        SUBMITTED,
+        VALID,
+        INVALID,
+        ACCEPTED,
+        REJECTED,
+        CORRECT,
+        INCORRECT,
+        UNVERIFIABLE
+    }
+
+    uint256 public constant SECONDS_IN_DAY = 86400;
+    uint256 public constant SECONDS_IN_MINUTE = 60;
+
+    uint256 public currentEpoch;
+    uint256 public epochDuration;
+    uint256 public constant MIN_STAKE = 1 ether;
+    uint256 public delta = 1 * SECONDS_IN_DAY;
+    uint256 public p = 1 * SECONDS_IN_MINUTE;
+    uint256 public supermajorityStake;
+    uint256 public epochStartTimestamp;
+    uint256 public totalStake;
```

Comment:
These look like internal members of the contract state, why do they need to be public?

Reply:
You can request public members via `methods.data()` (in most ETH client implementations). These are parameters which other users of the protocol might benefit from accessing as well.